### DISCUSSION TITLE
Controlling ARFLAGS when creating the static libs

### DISCUSSION THREAD

**Discussion by jmeekhof** [edenhill/librdkafka#3883] [Q&A] [answered]
I'm using this project as a dependency that I'm building with Bazel. I'm using the [rules_foreign_cc](https://bazelbuild.github.io/rules_foreign_cc/0.8.0/configure_make.html) `configure_make` rule to build librdkafka. I've made in 95% of the way and can compile everything. The process is dying when it attempts to create the static C++ object.
```sh
Creating static library librdkafka++.a
ar rcsrcsD  librdkafka++.a RdKafka.o ConfImpl.o HandleImpl.o ConsumerImpl.o ProducerImpl.o KafkaConsumerImpl.o TopicImpl.o TopicPartitionImpl.o MessageImpl.o HeadersImpl.o QueueImpl.o MetadataImpl.o
ar: two different operation options specified
```
Digging around a bit, the issue is with the ARFLAGS. The sandboxed environment that Bazel and rules_foreign_cc presents to the configure script cause the the `/Makefile.config` to generate a value of `ARFLAGS= rcsD`. This causes an issue with [line 118 of /mklove/Makefile.base](https://github.com/edenhill/librdkafka/blob/v1.6.1/mklove/Makefile.base#L118) because it's already issuing the `rcs` portion of the command.

**My question is**: How can I control the generation of the ARFLAGS? 

My attempts so far have included:
* Adding `--ARFLAGS=` as an option to `configure`.
* Setting an env var of `ARFLAGS=`
* Defininging to make `ARFLAGS=`

So far I'm not having much luck.

**Comment by edenhill** [comment]
I think your analysis is correct.

Maybe we should change it so that a user-supplied ARFLAGS (such as in your case) replaces, rather than adds to, the mklove-defined ARFLAGS, but that risks doing another operation than intended.

Another alternative is to remove duplicate operations from ARFLAGS, which is the actual problem here - `r` being specified twice, which is a bit tricky with bash - so maybe just remove duplicate `r`s?

**Reply by jmeekhof** [reply]
My next attempt was going to `sed` the `/Makefile.config` and remove my `ARFLAGS`. A little heavy handed, but I'm running out of ideas.

**Comment by jmeekhof** [comment, marked as answer]
@edenhill I found a workable solution. I cribbed this solution from [envoy](https://github.com/envoyproxy/envoy/blob/main/bazel/foreign_cc/BUILD). Basically, rules_foreign_cc allows you to specify Make targets, and with that command you can inject an environment variable.
```
    targets = [
	    "ARFLAGS='' libs install-subdirs",
    ]
```
I feel like they may be a safer way to do this, but this works for now.
